### ISSUE TITLE
Wrong error `non-type template argument refers to subobject` for extern arrays

### ISSUE THREAD

**Issue by llvmbot** [llvm/llvm-project#28513]
|  |  |
| --- | --- |
| Bugzilla Link | [28139](https://llvm.org/bz28139) |
| Version | 3.8 |
| OS | Windows NT |
| Reporter | LLVM Bugzilla Contributor |

## Extended Description 
code:
```cpp
template<int const * pci> struct X {};

extern int const ai[];
X<ai> xi;

int const ai[] = {0,1,2,3};
```

Then compiled with `clang++ -std=c++1z test.cpp`

Error `test.cpp:4:3: error: non-type template argument refers to subobject '&ai'` is reported.

**Comment by AaronBallman** [comment]
This issue was fixed in Clang 6: https://godbolt.org/z/f6dxsWYsv

**Comment by llvmbot** [comment]
@llvm/issue-subscribers-clang-frontend

Author: None (llvmbot)

<details>
|  |  |
| --- | --- |
| Bugzilla Link | [28139](https://llvm.org/bz28139) |
| Version | 3.8 |
| OS | Windows NT |
| Reporter | LLVM Bugzilla Contributor |

## Extended Description 
code:
```cpp
template&lt;int const * pci&gt; struct X {};

extern int const ai[];
X&lt;ai&gt; xi;

int const ai[] = {0,1,2,3};
```

Then compiled with `clang++ -std=c++1z test.cpp`

Error `test.cpp:4:3: error: non-type template argument refers to subobject '&amp;ai'` is reported.
</details>
